files/get-random-osd.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import subprocess
+import json
+import argparse
+import random
+import socket
+
+_is_rack = False
+
+def get_map():
+    cmd = ['ceph', 'osd', 'crush', 'tree', '--format', 'json']
+    ceph_osd_crush_tree_json = subprocess.check_output(cmd)
+    ceph_osd_crush_tree = json.loads(ceph_osd_crush_tree_json)
+
+    osd_map = {}
+    global _is_rack
+
+    for top_item in ceph_osd_crush_tree:
+        if top_item['type'] == 'root':
+            for l2_item in top_item['items']:
+                if l2_item['type'] == 'rack':
```

Comment:
what if we have l2_item['type' ] == 'pod'  and  l3_item['type'] == 'rack'  ?  this may occurs in UOS4.0,  so I think we should recursively go through the crush_tree and find out which layer is rack and which layer is host